core/core-api/src/main/java/com/nerd/favorite18/core/api/ranking/controller/RankRadisController.java

```diff
@@ -0,0 +1,37 @@
+package com.nerd.favorite18.core.api.ranking.controller;
+
+import com.nerd.favorite18.core.api.ranking.service.RedisRankService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+
+@Slf4j
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/songs/rank")
+@RestController
+public class RankRadisController {
+
+    private final RedisRankService redisRankService;
+
+    @GetMapping("/redis/rank/{songId}")
+    public String getSongCount(@PathVariable Long songId) {
+        return redisRankService.getSongCounts(songId);
+    }
+
+    @PostMapping("/redis/rank")
+    public void setSongCount(@RequestBody HashMap<String, String> body) {
+        Long songId = Long.valueOf(body.get("songId").toString());
+        String searchCnt = body.get("searchCount").toString();
+        redisRankService.setSongCounts(songId, searchCnt);
+    }
+
+    @PostMapping("/redis/rank/increment/{songId}")
+    public ResponseEntity<String> incrementSongCount(@PathVariable Long songId) {
+        // 여기서는 간단히 조회수를 1 증가시킨다고 가정
+        redisRankService.incrementSongCount(songId);
+        return ResponseEntity.ok("Song count incremented for songId: " + songId);
```

Comment:
이런건 잡담 인데요, 저는 이런방식을 선호합니다. 다양한 방식을 알고 계시면 좋을 듯 해서 기록해둡니당 

참고로 문자열은 따로 변수같은걸로 빼서 많이 해둡니다. 그래서 보통 변수명이 들어가는 편이에용 :)
```suggestion
        return ResponseEntity.ok(String.format("Song count incremented for songId: %d", songId));
```